src/model_theory/terms_and_formulas.lean

```diff
@@ -249,6 +249,23 @@ instance : has_sup (L.bounded_formula α n) := ⟨λ f g, f.not.imp g⟩
 /-- The biimplication between two bounded formulas. -/
 protected def iff (φ ψ : L.bounded_formula α n) := φ.imp ψ ⊓ ψ.imp φ
 
+protected theorem induction_on {P : ∀ {n}, L.bounded_formula α n → Prop}
```

Comment:
Is the only difference with `bounded_formula.rec_on` that you use the notation `⊥`? Do you need this? 
Please mention the differences in a doc string.
Can you replace `Prop` by `Sort*`?

Reply:
It does work slightly better in some proofs in #12558. I will go through and check there which induction principles actually get used in term mode, as I doubt the rest are necessary.